### PR TITLE
Remove unused parameters for AnalysisRegistry#processAnalyzerFactory

### PR DESCRIPTION
**Pull request by liketic**
Remove unused parameters for ```AnalysisRegistry#processAnalyzerFactory``` and ```AnalysisRegistry#processNormalizerFactory```.  Any comments are appreciated. 👍

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java [hunk 1]
```diff
@@ -188,9 +183,9 @@ public IndexAnalyzers build(IndexSettings indexSettings) throws IOException {
     }
 
     public Map<String, AnalyzerProvider<?>> buildNormalizerFactories(IndexSettings indexSettings) throws IOException {
-        final Map<String, Settings> noralizersSettings = indexSettings.getSettings().getGroups("index.analysis.normalizer");
+        final Map<String, Settings> normalizersSettings = indexSettings.getSettings().getGroups("index.analysis.normalizer");
```
Comment (liketic) [inline core/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java]: Fix a typo.

File: core/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java [hunk 2]
```diff
@@ -510,11 +491,9 @@ public IndexAnalyzers build(IndexSettings indexSettings,
             unmodifiableMap(analyzers), unmodifiableMap(normalizers));
     }
 
-    private void processAnalyzerFactory(DeprecationLogger deprecationLogger,
-                                        IndexSettings indexSettings,
+    private void processAnalyzerFactory(IndexSettings indexSettings,
```
Comment (liketic) [inline core/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java]: Remove unused ```deprecationLogger``` and ```indexSettings``` and ```analyzerAliases ```.

File: core/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java [hunk 3]
```diff
@@ -455,33 +450,20 @@ public IndexAnalyzers build(IndexSettings indexSettings,
...
-            processAnalyzerFactory(deprecationLogger, indexSettings, entry.getKey(), entry.getValue(), analyzerAliases, analyzers,
+            processAnalyzerFactory(indexSettings, entry.getKey(), entry.getValue(), analyzers,
                 tokenFilterFactoryFactories, charFilterFactoryFactories, tokenizerFactoryFactories);
         }
         for (Map.Entry<String, AnalyzerProvider<?>> entry : normalizerProviders.entrySet()) {
-            processNormalizerFactory(deprecationLogger, indexSettings, entry.getKey(), entry.getValue(), normalizers,
+            processNormalizerFactory(entry.getKey(), entry.getValue(), normalizers,
                     tokenizerFactoryFactories.get("keyword"), tokenFilterFactoryFactories, charFilterFactoryFactories);
         }
-        for (Map.Entry<String, NamedAnalyzer> entry : analyzerAliases.entrySet()) {
```
Comment (liketic) [inline core/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java]: ```analyzerAliases``` is never updated.

**Review by cbuescher — APPROVED** [review]
@liketic thanks for cleaning this up, it looks like most if these were leftovers from removing the 2.0 backward compatibility layer in #26245. 
The change looks good to me, I will run tests and merge once they pass.

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@liketic CI doesn't seem happy, although the error doesn't seem to be related to your changes, could you either rebase your branch or merge in master before I kick off another build?

**Comment by liketic** [comment]
@cbuescher Rebase done. Please try again.

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Review by cbuescher — COMMENTED** [review]
Thanks, I will merge this in

**Pull request merged**